staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -156,6 +170,11 @@ func (a customResourceStrategy) Validate(ctx context.Context, obj runtime.Object
 
 		// validate x-kubernetes-list-type "map" and "set" invariant
 		errs = append(errs, structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], u.Object)...)
```

Comment:
unrelated to this PR, but noticed the embedded resources and x-kubernetes-list-type "map" and "set" invariant validation here wasn't happening in `statusStrategy#ValidateUpdate` ... does that mean embedded objects and list-type fields under status can contain invalid data? if so, let's open an issue to track that and fix it in a ratcheting way

Reply:
opened https://github.com/kubernetes/kubernetes/issues/106401